Source/Core/DolphinWX/FrameTools.cpp

```diff
@@ -862,11 +848,31 @@ void CFrame::DoStop()
     if (NetPlayDialog::GetNetPlayClient())
       NetPlayDialog::GetNetPlayClient()->Stop();
 
+    if (!m_tried_graceful_shutdown && TriggerSTMPowerEvent())
+    {
+      m_tried_graceful_shutdown = true;
+      return;
+    }
     Core::Stop();
     UpdateGUI();
   }
 }
 
+bool CFrame::TriggerSTMPowerEvent()
+{
+  const auto& stm = WII_IPC_HLE_Interface::GetDeviceByName("/dev/stm/eventhook");
```

Comment:
Since I was the original code author, I think it'd be a good idea to fix this :)
Fixed.